tool/tsh/common/proxy.go

```diff
@@ -567,9 +569,51 @@ func onProxyCommandApp(cf *CLIConf) error {
 		addr = fmt.Sprintf("127.0.0.1:%s", cf.LocalProxyPort)
 	}
 
-	listener, err := net.Listen("tcp", addr)
-	if err != nil {
-		return trace.Wrap(err)
+	// Declare the listener
+	var listener net.Listener
+
+	// Determine if we should use TLS for the local listener
+	if cf.LocalTLS {
+		var serverCert tls.Certificate
+
+		// Attempt to load the certificate from given certificate files.
+		if cf.LocalTLSCertificate != "" && cf.LocalTLSKey != "" {
+			serverCert, err = tls.LoadX509KeyPair(cf.LocalTLSCertificate, cf.LocalTLSCertificate)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+		} else {
+			log.Debug("Generating self-signed certificate for local port.")
+
+			// Generate a self-signed certificate to use in the proxy.
+			signedCert, err := cert.GenerateSelfSignedCert([]string{"localhost"}, nil, x509.ExtKeyUsageAny)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+
+			// Combine the key and certificate into a usable object for the Listener.
+			serverCert, err = tls.X509KeyPair(signedCert.Cert, signedCert.PrivateKey)
+			if err != nil {
+				return trace.Wrap(err)
+			}
```

Comment:
I really like the idea of auto-generating self-signed cert. The problem with the current approach is that the CA is not exported so the client has to use insecure mode (e.g `curl -k`) in order to connect.

`tsh` already has a way to generate self-signed ca for apps and it's saved in tsh keys dir. What do you think switch to that and print out the CA path. For example:
```suggestion
			profile, err := cf.ProfileStatus()  
			if err != nil {   
				return trace.Wrap(err)
			}                                                           
			serverCert, err = loadAppSelfSignedCA(profile, tc, cf.AppName)
			if err != nil { 
				return trace.Wrap(err)                                                                                                                             
			}                                         
			fmt.Printf("Generated self-signed certficate at %v.\n\n", profile.AppLocalCAPath(cf.AppName))
```

Then you can use this CA path for the clients:
```bash
$ tsh proxy app --tls some-app -p 8888
Generated self-signed certficate at /Users/stevehuang/.tsh/keys/teleport.dev.aws.stevexin.me/STeve-app/teleport.dev.aws.stevexin.me/some-app-localca.pem.

Proxying connections to some-app on 127.0.0.1:8888
$ curl --cacert /Users/stevehuang/.tsh/keys/teleport.dev.aws.stevexin.me/STeve-app/teleport.dev.aws.stevexin.me/some-app-localca.pem https://localhost:8888
```

Let me know what you think!